### PR TITLE
refactor: use `getLocFromIndex` in `no-invalid-label-refs`

### PR DESCRIPTION
**Pull request by lumirlumir**
## Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/.github/blob/master/CONTRIBUTING.md).

## What is the purpose of this pull request?

This PR originated from #536. While working on that, I found the scope had become too large, so I split this refactoring into a separate PR.

In this PR, I've used the native `getLocFromIndex` method added in #376 instead of the custom `findOffsets` util.

Notable changes:

- Replace the `findOffsets` util with the native `getLocFromIndex`.
- Use ESQuery selectors to reduce duplicated logic.

## What changes did you make? (Give an overview)

In this PR, I've used the native `getLocFromIndex` method added in #376 instead of the custom `findOffsets` util.

## Related Issues

Ref: #376, #536

#### Is there anything you'd like reviewers to focus on?

N/A

### CODE REVIEW

File: src/rules/no-invalid-label-refs.js [hunk 1]
```diff
@@ -17,33 +17,30 @@ import { findOffsets, illegalShorthandTailPattern } from "../util.js";
  * @import { Position } from "unist";
  * @import { Text } from "mdast";
  * @import { MarkdownRuleDefinition } from "../types.js";
+ * @import { MarkdownSourceCode } from "../index.js";
```
Comment (lumirlumir) [inline src/rules/no-invalid-label-refs.js]: This change was motivated by https://github.com/eslint/markdown/pull/366#discussion_r2084820948.

The `Parameters<MarkdownRuleDefinition['create']>[0]['sourceCode']` expression was needed due to a collision between `rollup` and `tools/dedupe-types` at the time, but since `@eslint/markdown` no longer uses `rollup`, it's now safe to use `../index.js`.

File: tests/rules/no-invalid-label-refs.test.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ ruleTester.run("no-invalid-label-refs", rule, {
 		"![foo][]\n\n[foo]: http://bar.com/image.jpg",
 		"[  foo ][]\n\n[foo]: http://bar.com/image.jpg",
 		"[eslint][\n\n]",
+		"[*eslint*][]\n\n[*eslint*]: http://bar.com",
```
Comment (lumirlumir) [inline tests/rules/no-invalid-label-refs.test.js:35]: Using strong or emphasis inside a label is an easy-to-forget edge case and often leads to unintended side effects when the check is missed, so I've added one more test.

This is a valid label according to the [AST](https://explorer.eslint.org/#eslint-explorer=IntcInN0YXRlXCI6e1widG9vbFwiOlwiYXN0XCIsXCJjb2RlXCI6e1wiamF2YXNjcmlwdFwiOlwie1xcbiAgXFxcImZvb1xcXCI6IFxcXCJmb29cXFwiLFxcbn1cIixcImpzb25cIjpcIntcXG4gIFxcXCJhXFxcIjogXFxcImJcXFwiXFxufVwiLFwibWFya2Rvd25cIjpcIlsqZXNsaW50Kl1bXVxcblxcblsqZXNsaW50Kl06IGh0dHA6Ly9iYXIuY29tXCIsXCJjc3NcIjpcImEgeyBjb2xvcjogcmVkICEvKiBjb21tZW50ICovaW1wb3J0YW50OyB9XCIsXCJodG1sXCI6XCI8IURPQ1RZUEUgaHRtbD5cXG48IS0tXFxuVHlwZSBvciBwYXN0ZSBzb21lIEhUTUwgaGVyZSB0byBsZWFybiBtb3JlIGFib3V0XFxudGhlIHN0YXRpYyBhbmFseXNpcyB0aGF0IEVTTGludCBjYW4gZG8gZm9yIHlvdS5cXG5cXG5UaGUgdGFicyBhcmU6XFxuXFxuLSBBU1QgLSBUaGUgQWJzdHJhY3QgU3ludGF4IFRyZWUgb2YgdGhlIGNvZGUsIHdoaWNoIGNhblxcbmJlIHVzZWZ1bCB0byB1bmRlcnN0YW5kIHRoZSBzdHJ1Y3R1cmUgb2YgdGhlIGNvZGUuIFlvdVxcbmNhbiB2aWV3IHRoaXMgc3RydWN0dXJlIGFzIEpTT04gb3IgaW4gYSB0cmVlIGZvcm1hdC5cXG4tLT5cXG5cXG48aHRtbCBsYW5nPVxcXCJlblxcXCI%2BXFxuICAgIDxoZWFkPlxcbiAgICAgICAgPG1ldGEgY2hhcnNldD1cXFwiVVRGLThcXFwiPlxcbiAgICAgICAgPHRpdGxlPkhUTUw8L3RpdGxlPlxcbiAgICA8L2hlYWQ%2BXFxuICAgIDxib2R5PlxcbiAgICAgICAgPHA%2BVGV4dDwvcD5cXG4gICAgPC9ib2R5PlxcbjwvaHRtbD5cIn0sXCJsYW5ndWFnZVwiOlwibWFya2Rvd25cIixcImpzT3B0aW9uc1wiOntcInBhcnNlclwiOlwiZXNwcmVlXCIsXCJzb3VyY2VUeXBlXCI6XCJtb2R1bGVcIixcImVzVmVyc2lvblwiOlwibGF0ZXN0XCIsXCJpc0pTWFwiOnRydWV9LFwianNvbk9wdGlvbnNcIjp7XCJqc29uTW9kZVwiOlwianNvblwiLFwiYWxsb3dUcmFpbGluZ0NvbW1hc1wiOmZhbHNlfSxcImNzc09wdGlvbnNcIjp7XCJjc3NNb2RlXCI6XCJjc3NcIixcInRvbGVyYW50XCI6ZmFsc2V9LFwibWFya2Rvd25PcHRpb25zXCI6e1wibWFya2Rvd25Nb2RlXCI6XCJnZm1cIixcIm1hcmtkb3duRnJvbnRtYXR0ZXJcIjpcIm9mZlwifSxcImh0bWxPcHRpb25zXCI6e1widGVtcGxhdGVFbmdpbmVTeW50YXhcIjpcInR3aWdcIixcImZyb250bWF0dGVyXCI6ZmFsc2V9LFwid3JhcFwiOnRydWUsXCJ2aWV3TW9kZXNcIjp7XCJhc3RWaWV3XCI6XCJ0cmVlXCIsXCJzY29wZVZpZXdcIjpcImZsYXRcIixcInBhdGhWaWV3XCI6XCJncmFwaFwifSxcInBhdGhJbmRleFwiOntcImluZGV4XCI6MCxcImluZGV4ZXNcIjoxfSxcImVzcXVlcnlTZWxlY3RvclwiOntcInNlbGVjdG9yXCI6XCJcIn19LFwidmVyc2lvblwiOjB9Ig%3D%3D):

image

**Comment by lumirlumir** [comment]
It seems there is a problem on the macOS runner:

https://www.githubstatus.com/

image

File: src/rules/no-invalid-label-refs.js (hunk 1 above)
Comment (mdjermanovic) [inline src/rules/no-invalid-label-refs.js]:
```suggestion
 * @import { MarkdownSourceCode } from "../language/markdown-source-code.js";
```

It might be better this way, to avoid circular references?

Comment (lumirlumir) [reply src/rules/no-invalid-label-refs.js]: Sure, I've updated it in https://github.com/eslint/markdown/pull/546/commits/ae1b8878b332d8c8ee568d7d80ff6b5134274527 👍
